src/vm/arm64/crthelpers.asm

```diff
@@ -12,85 +12,13 @@
 
     TEXTAREA
 
-; Calls to JIT_MemSet is emitted by jit for initialization of large structs. 
-; We need to provide our own implementation of memset instead of using the ones in crt because crt implementation does not gurantee 
-; that aligned 8/4/2 - byte memory will be written atomically. This is required because members in a struct can be read atomically 
-; and their values should be written atomically.
-; 
-;
 ;void JIT_MemSet(void *dst, int val, SIZE_T count)
 ;{
 ;    uint64_t valEx = (unsigned char)val;
 ;    valEx = valEx | valEx << 8;
 ;    valEx = valEx | valEx << 16;
 ;    valEx = valEx | valEx << 32;
 ;
-;    size_t dc_zva_size = 4ULL << DCZID_EL0.BS;
-;
-;    uint64_t use_dc_zva = (val == 0) && !DCZID_EL0.p ? count / (2 * dc_zva_size) : 0; // ~Minimum size (assumes worst case alignment)
-;
-;    // If not aligned then make it 8-byte aligned   
-;    if(((uint64_t)dst&0xf) != 0)
-;    {
-;        // Calculate alignment we can do without exceeding count
-;        // Use math to avoid introducing more unpredictable branches
-;        // Due to inherent mod in lsr, ~7 is used instead of ~0 to handle count == 0
-;        // Note logic will fail is count >= (1 << 61).  But this exceeds max physical memory for arm64
-;        uint8_t align = (dst & 0x7) & (~uint64_t(7) >> (countLeadingZeros(count) mod 64))
```

Comment:
There were two bugs in the pseudo code for calculating align.  The correct calculation should have been
```
uint8_t align = (8 - (dst & 0x7)) & (~uint64_t(7) >> (countLeadingZeros(count) % 64))
```
This escaped due to incomplete unit testing.  Test coverage also failed to hit the problematic corners.